akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1627,17 +1627,36 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
 
 /** DO NOT INHERIT: Class is open only for use by akka-slf4j*/
 @DoNotInherit
-class LogMarker(val name: String)
+class LogMarker(val name: String, val properties: Map[String, Any]) {
+
+  // for binary compatibility
+  def this(name: String) = this(name, Map.empty)
+
+  /** Java API */
+  def getProperties: java.util.Map[String, Object] = {
+    import scala.collection.JavaConverters._
+    properties.map { case (k, v) => (k, v.asInstanceOf[AnyRef]) }.asJava
+  }
+}
+
 object LogMarker {
 
   /** The Marker is internally transferred via MDC using using this key */
   private[akka] final val MDCKey = "marker"
 
-  def apply(name: String): LogMarker = new LogMarker(name)
+  def apply(name: String): LogMarker = new LogMarker(name, Map.empty)
```

Comment:
Three lines up, um, so on top of transferring these values through markers to write them as MDC we also transfer markers as a MDC key? Quite confusing.

Reply:
no, that MDCKey is something old that isn't used. deprecated and `LogMarker.extractFromMDC` kept for binary compatibility, I guess. 

When logging with marker they are published as `LogEventWithMarker`.

Reply:
Good